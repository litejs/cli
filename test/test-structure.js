
describe("TestSuite 1 with fn", function() {
	Object.prototype.foo = 1
	describe.onend = function() {
		delete Object.prototype.foo
	}
	this
	.test("it should run first test 1.1", function(assert) {
		assert.ok(true)
		assert.ok(true)
		.notOk(false)
		setTimeout(assert.end, 1)
	})
})
.describe("TestSuite 2 with map", {
	"Case 2.1": function() {
		this.plan(1).ok(1)
	},
	"Case 2.2": function(assert, mock) {
		this.ok(1).plan(1)
	},
	"Nested Suite 2.3": {
		"Case 2.3.1": null,
		"Case 2.3.2": null
	},
	"Nested Suite 2.4": {
		"Case 2.4.1": null,
		"Case 2.4.2": null
	},
	"Case 2.5": null
})


var suitesTable = [
	[ "env.1", function() { return Promise.resolve(1) } ],
	[ "env.2", function() { return Promise.resolve(2) } ]
]
, casesTable = [
	[ "ca1", 3 ],
	[ "ca2", 4 ]
]

describe("TestSuite 3 {0}", suitesTable, function(suiteName, suiteSetup) {
	it("set up a " + suiteName, suiteSetup)
	it("tests case {0}", casesTable, function(caseName, caseId, assert) {
		assert.type(caseId, "number")
		assert.type(suiteName, "string")
		assert.type(caseName, "string")
		assert.end()
	})
})

describe("TestSuite 4")
.test("it should run chained test 4.1", function(assert) {
	assert.plan(1)
	assert.ok(true)
})
.it("supports a pending test 4.2")
.it("_should skip test by name 4.3", function(assert) {
	assert.fail()
	assert.end()
})

