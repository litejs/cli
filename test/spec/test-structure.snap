TestSuite 1 with fn
  ✔ 1. it should run first test 1.1 [3/3]
TestSuite 2 with map
  ✔ 2. Case 2.1 [1/1]
  ✔ 3. Case 2.2 [1/1]
  Nested Suite 2.3
    Case 2.3.1
    Case 2.3.2
  Nested Suite 2.4
    Case 2.4.1
    Case 2.4.2
  Case 2.5
TestSuite 3 env.1
  ✔ 4. it set up a env.1 [0/0]
  ✔ 5. it tests case ca1 [3/3]
  ✔ 6. it tests case ca2 [3/3]
TestSuite 3 env.2
  ✔ 7. it set up a env.2 [0/0]
  ✔ 8. it tests case ca1 [3/3]
  ✔ 9. it tests case ca2 [3/3]
TestSuite 4
  ✔ 10. it should run chained test 4.1 [1/1]
  ∅ 11. it supports a pending test 4.2
  ∅ 12. it _should skip test by name 4.3
1..12
# pass  12/12 [18/18]
# skip  2
